source/common/http/header_utility.cc

```diff
@@ -235,7 +235,96 @@ bool HeaderUtility::headerNameContainsUnderscore(const absl::string_view header_
   return header_name.find('_') != absl::string_view::npos;
 }
 
+namespace {
+// This function is used to validate the authority as is done in nghttp2 (based on
```

Comment:
I think it would be fine to omit references to nghttp2. My suggestion would be:

s/as is done in nghttp2[^.]+//

Reply:
I agree that removing the nghttp2 reference entirely makes sense, and this should be Envoy-specific view of the validation. I rephrased the comment, and decided to intentionally mention oghttp2, so in the future it can be used if needed (leaving a bread crumb).

Thanks!